scipy/_lib/_util.py

```diff
@@ -521,6 +522,29 @@ def _fixed_default_rng(seed=1638083107694713882823079058616272161):
         np.random.default_rng = orig_fun
 
 
+def _rng_html_rewrite(func):
+    """Rewrite the HTML rendering of ``np.random.default_rng``.
+
+    This is intended to decorate
+    ``numpydoc.docscrape_sphinx.SphinxDocString._str_examples``.
+
+    Examples are only run by Sphinx when there are plot involved. Even so,
+    it does not change the result values getting printed.
```

Comment:
Could you elaborate (just so I can understand)? Why is the comment important?

Reply:
This is to tell the reader that even when the docstrings is being run by a Sphinx process, no return value is being changed. e.g. if the docstrings in the code is the following

```python
>>> rng.random()
0.7
```

When Sphinx renders that, even if the output of `rng.random()` changes to something else than 0.7 if will still print 0.7. There is no rewrite of what is going to be printed.

The only time Sphinx actually execute any docstrings is when there are plots, otherwise nothing is run. When there is a plot, the code is being run, and the random stuff can change as we have the rewrite of the seed. So plots would be able to change, but the printed info that we wrote are not being touched.

Reply:
So: 
- The rendered text (e.g. English, code, code output) will be the same as in the `.py` file (except for the removal of the seed number)
- The rendered plots *will* be different compared to if running the code in the `.py` file if they depend on random data. This is because Sphinx produces these when it runs the code with `default_rng()`; Sphinx never sees the seed you specified.

If so, it sounds like plotting side effect will be new. Are we sure that there are no plots in the documentation that depend meaningfully on the random seed from which they were produced?

Reply:
Note that the plotting situation would be strictly similar as right now.

We could probably also intercept the plotting logic and do some magic.

Reply:
Ok. I just didn't want this to break any new examples (that aren't already broken by text that doesn't agree with the plots).

Reply:
Yes, I see that plots are currently produced without a fixed seed. See e.g. `ppcc_pot` histogram in

[1.9.3](https://docs.scipy.org/doc/scipy-1.9.3/reference/generated/scipy.stats.ppcc_plot.html)
![image](https://user-images.githubusercontent.com/6570539/220207491-0578a1a2-99f5-45c0-bbaf-70df1354e8d6.png)

[1.10.0](https://docs.scipy.org/doc/scipy-1.10.0/reference/generated/scipy.stats.ppcc_plot.html)
![image](https://user-images.githubusercontent.com/6570539/220207534-9c8ed747-ffab-4066-aab8-7ee0370f6be4.png)

[1.10.1](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.ppcc_plot.html#scipy.stats.ppcc_plot)
![image](https://user-images.githubusercontent.com/6570539/220207614-78cea388-2a2a-471e-9996-bf88a5f36050.png)